src/gc.c

```diff
@@ -665,59 +721,65 @@ static inline int maybe_collect(void)
 
 // preserved values
 
-static arraylist_t preserved_values;
-
 DLLEXPORT int jl_gc_n_preserved_values(void)
 {
-    return preserved_values.len;
+    FOR_CURRENT_HEAP
+        return preserved_values.len;
+    END
 }
 
 DLLEXPORT void jl_gc_preserve(jl_value_t *v)
 {
-    arraylist_push(&preserved_values, (void*)v);
+    FOR_CURRENT_HEAP
+        arraylist_push(&preserved_values, (void*)v);
+    END
 }
 
 DLLEXPORT void jl_gc_unpreserve(void)
 {
-    (void)arraylist_pop(&preserved_values);
+    FOR_CURRENT_HEAP
+        (void)arraylist_pop(&preserved_values);
+    END
 }
 
 // weak references
 
-static arraylist_t weak_refs;
-
 DLLEXPORT jl_weakref_t *jl_gc_new_weakref(jl_value_t *value)
 {
     jl_weakref_t *wr = (jl_weakref_t*)alloc_1w();
     jl_set_typeof(wr, jl_weakref_type);
     wr->value = value;
-    arraylist_push(&weak_refs, wr);
+    FOR_CURRENT_HEAP
+        arraylist_push(&weak_refs, wr);
+    END
     return wr;
 }
 
 static void sweep_weak_refs(void)
 {
-    size_t n=0, ndel=0, l=weak_refs.len;
-    jl_weakref_t *wr;
-    void **lst = weak_refs.items;
-    void *tmp;
+    FOR_CURRENT_HEAP
+        size_t n=0, ndel=0, l=weak_refs.len;
+        jl_weakref_t *wr;
+        void **lst = weak_refs.items;
+        void *tmp;
 #define SWAP_wr(a,b) (tmp=a,a=b,b=tmp,1)
-    if (l == 0)
-        return;
-    do {
-        wr = (jl_weakref_t*)lst[n];
-        if (gc_marked(jl_astaggedvalue(wr))) {
-            // weakref itself is alive
-            if (!gc_marked(jl_astaggedvalue(wr->value)))
-                wr->value = (jl_value_t*)jl_nothing;
-            n++;
-        }
-        else {
-            ndel++;
-        }
-    } while ((n < l-ndel) && SWAP_wr(lst[n],lst[n+ndel]));
+        if (l == 0)
+            return;
+        do {
+            wr = (jl_weakref_t*)lst[n];
+            if (gc_marked(jl_astaggedvalue(wr))) {
+                // weakref itself is alive
+                if (!gc_marked(jl_astaggedvalue(wr->value)))
+                    wr->value = (jl_value_t*)jl_nothing;
+                n++;
+            }
+            else {
+                ndel++;
+            }
+        } while ((n < l-ndel) && SWAP_wr(lst[n],lst[n+ndel]));
```

Comment:
The problem is convincing Github that the tabs are not there.   When I check out my branch, `src/gc.c` has no tabs.